### PR TITLE
feat: [IDP-1437]: IDP Cookiecutter step schema changes

### PR DESCRIPTION
**Pull request by devesh-harness**
-> This is for adding the schema changes for idp cookicutter step in idp-stage.
-> Change is tested in local and is working as excpectd.

### CODE REVIEW

File: v0/pipeline/steps/idp/idpcookiecutter-node.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+type: object
+required:
+  - identifier
+  - name
+  - spec
+  - type
+properties:
+  description:
+    type: string
+    desc: This is the description for IACMApprovalStepNode
```
Comment (vikyathharekal) [inline v0/pipeline/steps/idp/idpcookiecutter-node.yaml:11]:
```suggestion
    desc: This is the description for IDPCookieCutterStepNode
```

File: v0/pipeline/steps/idp/idpcookiecutter-info.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+title: IDPCookieCutterPlanInfo
+allOf:
+  - $ref: ../../common/step-spec-type.yaml
+  - type: object
+    required:
+    - isPublicTemplate
+    - cookieCutterVariables
+    properties:
+      isPublicTemplate:
```
Comment (vikyathharekal) [inline v0/pipeline/steps/idp/idpcookiecutter-info.yaml:9]: Just confirming if we really need this? Or can we infer based on which one of publicTemplateUrl or pathForTemplate is set?

File: v1/pipeline/steps/idp/idpcookiecutter-node.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+type: object
+required:
+  - identifier
+  - name
+  - spec
+  - type
+properties:
+  description:
+    type: string
+    desc: This is the description for IACMApprovalStepNode
```
Comment (vikyathharekal) [inline v1/pipeline/steps/idp/idpcookiecutter-node.yaml:11]:
```suggestion
    desc: This is the description for IDPCookieCutterStepNode
```

File: v0/pipeline/steps/idp/idpcookiecutter-info.yaml (hunk 2 above)
Comment (devesh-harness) [reply v0/pipeline/steps/idp/idpcookiecutter-info.yaml:9]: I added it based on the discussion and thought if needed by UI, once UI integration is done if we don't need it, we will remove it. Correct from the backend we can handle it without this variable as well.

**Review by vikyathharekal — APPROVED** [review]

**Review by archit-harness — APPROVED** [review]

**Review by utkarshchoubey-harness — APPROVED** [review]

**Review by sandeepa-kv-jois — APPROVED** [review]

**Pull request merged**